api/index.js

```diff
@@ -1,7 +1,31 @@
 'use strict'
 var path = require('path');
+var logger = require('./lib/log/logger');
 var args = process.argv;
 if (!(args.length == 4 && args[2] == "-c" && args[3] != "")) {
     throw new Error("missing config file\nUsage:use '-c' option to specify the config file path");
 }
 var apiServer = require(path.join(__dirname, 'app.js'))(args[3]);
+
+var gracefulShutdown = function(signal) {
+  logger.info("Received " + signal + " signal, shutting down gracefully...");
+  apiServer.shutdown(function() {
+    logger.info('Everything is cleanly shutdown');
+  })
+}
+
+//listen for SIGTERM signal e.g. kill
+process.on ('SIGTERM', function(){
```

Comment:
We shouldn't be doing a graceful shutdown on SIGTERM. That should be the escape valve for kill immediate.